### DISCUSSION TITLE
implementing-dijkstras-algorithm-for-finding-the-shortest-path-between-two-nodes-using-priorityqueue-in-dotnet-9/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#414] [Announcements]
# implementing-dijkstras-algorithm-for-finding-the-shortest-path-between-two-nodes-using-priorityqueue-in-dotnet-9/

In this post I describe Dijkstra's algorithm for finding the shortest path between two nodes and how you can implement it with the PriorityQueue type in .NET 9

https://andrewlock.net/implementing-dijkstras-algorithm-for-finding-the-shortest-path-between-two-nodes-using-priorityqueue-in-dotnet-9/

**Comment by viceroypenguin** [comment]
By the way, [SuperLinq](https://github.com/viceroypenguin/SuperLinq) has an implementation of both Dijkstra and A*. I took the bcl implementation of `PriorityQueue`, and modified it so that the priority could be updated in O(log N); downside is it doubles the memory cost. 

Feel free to see the documentation on these methods:
* point to point Dijkstra and A*: https://viceroypenguin.github.io/SuperLinq/api/SuperLinq.SuperEnumerable.GetShortestPath.html
* full-map Dijkstra: https://viceroypenguin.github.io/SuperLinq/api/SuperLinq.SuperEnumerable.GetShortestPaths.html
